src/qml/qmlwaveformrenderer.cpp

```diff
@@ -207,14 +212,43 @@ QmlWaveformRendererFactory::Renderer QmlWaveformRendererMark::create(
                 pMark->textColor(),
                 pMark->align(),
                 pMark->text(),
-                pMark->pixmap(),
-                pMark->icon(),
+                pMark->pixmap().toLocalFile(),
+                pMark->icon().toLocalFile(),
                 pMark->color(),
-                priority)));
+                priority,
+                Cue::kNoHotCue,
+                {},
+                pMark->endPixmap().toLocalFile(),
+                pMark->endIcon().toLocalFile(),
+                pMark->disabledOpacity(),
+                pMark->enabledOpacity())));
         priority--;
     }
     const auto* pMark = defaultMark();
     if (pMark != nullptr) {
+        const QString pixmap = pMark->pixmap().toLocalFile();
+        const QString endPixmap = pMark->endPixmap().toLocalFile();
+        const QString icon = pMark->icon().toLocalFile();
+        const QString endIcon = pMark->endIcon().toLocalFile();
+        // FIXME: the following checks should be done on the WaveformMarker
+        // setter (depends of #14515)
+        if (!pixmap.isEmpty() && !QFileInfo(pixmap).exists()) {
```

Comment:
clazy + redundant empty check. Qt does it for us. 
```suggestion
        if (!QFileInfo::exists(pixmap)) {
```
Also in other places below.

Reply:
There are more places below you need to care for.